src/common.h

```diff
@@ -978,6 +988,32 @@ enum {
     STATUS_INVALID_ARGS = 121,
 };
 
+/// Replaces usages of int-as-bool to incorporate minus-one-on-failure behavior with a strictly-typed
```

Comment:
This seems like an unnecessary abstraction to me. I don't think we should introduce this Tristate type. It's not motivated by this commit.